### DISCUSSION TITLE
Chat message history in a group chat

### DISCUSSION THREAD

**Discussion by bobbiaditya** [microsoft/autogen#1877] [Q&A]
Hello,

I have set up a multi-agent conversation in a group chat. However, I've encountered an issue where it appears that each agent shares the same history of messages, contrary to my expectation that each agent would have its own distinct chat history involving only their interactions with the group chat manager.

I believe it would be beneficial to restrict an agent's chat history to only include its conversations with the group chat manager, as it will constrain the context that each agent has.

Could anyone advise if there's a way to implement such a limitation on the chat history for each agent? I have reviewed the documentation but haven't found any guidance on this matter.

Thank you in advance for your help.

**Comment by farrelmahendra** [comment]
Same issue here

**Comment by ekzhu** [comment]
Group chat shares the message histories among all participants. For your case, have you looked into nested chats and sequential chats? 

See examples: https://microsoft.github.io/autogen/docs/Examples

**Reply by bobbiaditya** [reply]
thanks for bringing it up! will look into it

**Reply by sonichi** [reply]
I've also suggested adding options to specify restrictions to the recipients of each message in group chat.
cc @joshkyh @qingyun-wu @LeoLjl

**Reply by bobbiaditya** [reply]
Is it possible to create a group chat within a groupchat?

let's say on the highest level of group chat A that is managed by Manager A we have:
Agent A, user proxy A and Manager B.

And then on Manager B:
Agent B, user proxy B.

Hence the history of manager B would not be shared in the group chat A.

for building this setup, would it be better to create it using the group chat and gropchat manager functionalities, rather then nested/sequential setup?
